engines/advancedDetector.cpp

```diff
@@ -187,17 +187,26 @@ DetectedGame AdvancedMetaEngineDetection::toDetectedGame(const ADDetectedGame &a
 	}
 
 	DetectedGame game(getName(), desc->gameId, title, desc->language, desc->platform, extra, ((desc->flags & (ADGF_UNSUPPORTED | ADGF_WARNING)) != 0));
-	game.hasUnknownFiles = adGame.hasUnknownFiles;
+	game.hasUnknownFiles = adGame.getHasUnknownFiles();
 	game.matchedFiles = adGame.matchedFiles;
 
+	Common::HashMap<Common::String, bool, Common::IgnoreCase_Hash, Common::IgnoreCase_EqualTo> usesResourceFork;
+
+	for (const ADGameFileDescription *fileDesc = desc->filesDescriptions; fileDesc->fileName; fileDesc++) {
```

Comment:
EDIT:
There is `parseFileDesc()` below. What about it?

ORIGINAL COMMENT:

I'd suggest turning it into a method, something like:

```C++
String processCheckSum(char *md5, uint32 &flags) const;
```

And that function would strip down all prefixes and return the extracted flags. And then, you just do:

```C++
uint32 flags = desc->flags;

md5 = processCheckSum(fileDesc->md5, &flags);
```

And that thing will provide the combined flags. That will make it extensible, because, for instance, having `t:` for the tail md5 could be quite helpful

Reply:
Will do